src/components/WelcomingModal.tsx

```diff
@@ -4,6 +4,12 @@ interface Props {
     onSetUsername: (username: string) => void
 }
 
+/**
+ * A container component that prompts the user to enter a username upon initial load of the application.
```

Comment:
because the logic regarding the initial load isnt happening here, this might be more fitting
```suggestion
 * A container component that prompts the user to enter a username.
```